pom.xml

```diff
@@ -454,6 +469,79 @@
         </plugins>
     </reporting>
     <profiles>
+        <!-- Enable source jars only if a corresponding safe license file is found - projects
+        need to opt in by having one of the matching license files. Because maven activations
+        don't support multiple files, we need to have a separate profile per license. Finally, to
+        support both single and multi-module projects, we need to check for the files relative to
+        both the project and its parent directory (in multi-module projects we're interested in
+        source jars in the child modules, so need to look one level up). This has some risk
+        (checking one level above the project/repo directory) and requires some redundancy, but
+        seems to work sufficiently well.
+        -->
+        <profile>
+            <id>attach-sources-apache</id>
+            <activation>
+                <file>
+                    <exists>LICENSE-Apache</exists>
+                </file>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-source-plugin</artifactId>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>attach-sources-apache-parent</id>
+            <activation>
+                <file>
+                    <exists>../LICENSE-Apache</exists>
```

Comment:
+1, i see that we check for if the license file exists however no necessarily check for the validity of the content of the file.